types/base.d.ts

```diff
@@ -120,5 +120,13 @@ export interface ContextualCallbackTestContext extends CallbackTestContext {
 	context: any;
 }
 
+export interface Macro<T, I, E> {
+	(t: T, input: I, expected: E): void;
+	title? (providedTitle: string, input: I, expected: E): string;
+}
+export type Macros<T, I, E> = Macro<T, I, E> | Macro<T, I, E>[];
+
 export function test(name: string, run: ContextualTest): void;
 export function test(run: ContextualTest): void;
+export function test<I, E> (name: string, run: Macros<ContextualTestContext, I, E>, input: I, expected: E): void;
+export function test<I, E> (run: Macros<ContextualTestContext, I, E>, input: I, expected: E): void;
```

Comment:
@novemberborn Can you provide an example on when to use 0 or 1 argument?

I believe the syntax would then become

`run?: Macros<ContextualTestContext, I, E>, input?: I, expected?: E`

Or is `test` just passing all arguments to the macro?

```js
function macro(t, foo, bar, baz, unicorn) {
    t.is(unicorn, 'unicorn');
}

test('2 + 2 === 4', macro, 'foo', 'bar', 'baz', 'unicorn');
```